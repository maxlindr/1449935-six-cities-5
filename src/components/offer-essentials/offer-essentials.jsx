import React from 'react';
import {capitalizeFirstLetter} from '../offer-card-utils';
import {offerPropTypes} from '../../prop-types';

function OfferEssentials(props) {
  const {type, bedrooms, maxAdults} = props.offer;

  return (
    <ul className="property__features">
      <li className="property__feature property__feature--entire">
        {capitalizeFirstLetter(type)}
      </li>
      <li className="property__feature property__feature--bedrooms">
        {bedrooms} Bedrooms
      </li>
      <li className="property__feature property__feature--adults">
        Max {maxAdults} adults
      </li>
    </ul>
  );
}

OfferEssentials.propTypes = {
  offer: offerPropTypes
};

export {OfferEssentials};
export default React.memo(OfferEssentials);
