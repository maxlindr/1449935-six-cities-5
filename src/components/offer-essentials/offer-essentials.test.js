import React from 'react';
import renderer from 'react-test-renderer';
import {OfferEssentials} from './offer-essentials';
import offer from '../../../__mocks__/offer-mock';

test(`OfferEssentials должен корректно отображаться`, () => {
  const component = renderer.create(
      <OfferEssentials offer={offer}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
