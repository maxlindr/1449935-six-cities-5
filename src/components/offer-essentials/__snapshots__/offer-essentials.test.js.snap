// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferEssentials должен корректно отображаться 1`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    House
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    3
     Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 
    4
     adults
  </li>
</ul>
`;
